src/components/Form/FormDatePicker.react.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+
+import * as React from "react";
+
+import FormSelect from "./FormSelect.react";
+import FormInputGroup from "./FormInputGroup.react";
+
+type Props = {|
+  children?: React.Node,
+  className?: string,
+  defaultDate: Date,
+  minYear: number,
+  maxYear: number,
+  format: string,
+  monthLabels: Array<string>,
+  onChange?: Function,
+|};
+
+type State = {|
+  currentDate: Date,
+|};
+
+type ChangeTypes = "mm" | "yyyy" | "dd";
+
+class FormDatePicker extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    monthLabels: [
```

Comment:
it appears you've made monthLabels a required prop, however you have a default value.  You should either make it not required or get rid of the default value.  Same goes for a couple of these other properties

Reply:
@AaronCoplan I'm having a problem with this one. If I set this fields as optional in props flow starts giving `prop might be undefined`. Probably related to https://github.com/facebook/flow/issues/1660#issuecomment-386619834

Reply:
@AaronCoplan also related https://flow.org/en/docs/react/components/#toc-using-default-props

Reply:
Ah okay, that works for me.  Not exactly how we have been doing it but this approach works too!! 👍